### PR TITLE
Add tips list api

### PR DESCRIPTION
**Pull request by hyifeng**

### CODE REVIEW

**Comment by hyifeng** [comment]
ref #23

File: packages/server/config/config.dev.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  mongo: {
+    url: "mongodb://localhost:27017",
+  },
+  server: {
+    port: 3213,
+  },
+};
```
Comment (wliyongfeng) [inline packages/server/config/config.dev.js:8]: Ignore this file

File: packages/scan/src/utils/index.js [hunk 2]
```diff
@@ -15,8 +15,19 @@ function isExtrinsicSuccess(events) {
   return events.some((e) => e.event.method === "ExtrinsicSuccess");
 }
 
+function computeTipValue(tips) {
+  if (!tips || !tips.length) {
+    return null;
+  }
+
+  tips = tips.slice();
+  tips.sort((a, b) => a[1] - b[1]);
```
Comment (wliyongfeng) [inline packages/scan/src/utils/index.js:24]: Not good to change the argument variable. How about following code?

```js
const sortedTips = [...tips].sort((a, b) => a[1] - b[1])
```

File: packages/scan/src/extrinsic/treasury/tip.js [hunk 3]
```diff
@@ -74,8 +76,26 @@ async function saveTipTimeline(hash, state, args, indexer, sort) {
     hash,
     args,
     state,
-    meta: meta.toJSON(),
+    meta,
   });
+
+  const medianValue = computeTipValue((meta && meta.tips) || []);
```
Comment (wliyongfeng) [inline packages/scan/src/extrinsic/treasury/tip.js:82]: Will `meta?.tips || []` work in your environment?

File: packages/scan/src/extrinsic/treasury/tip.js [hunk 4]
```diff
@@ -74,8 +76,26 @@ async function saveTipTimeline(hash, state, args, indexer, sort) {
...
+        medianValue,
+        meta,
+        state: {
+          indexer,
+          state,
+          args,
+        },
+      },
+    }
+  );
```
Comment (wliyongfeng) [inline packages/scan/src/extrinsic/treasury/tip.js:98]: Don't do tip update in this `saveTipTimeline` function. Better to extract another function for clean code.

File: packages/scan/src/events/treasury/tip.js [hunk 5]
```diff
@@ -26,34 +27,61 @@ async function handleTipEvent(method, jsonData, indexer, sort) {
...
+      $set: {
+        ...updates,
+        state: {
+          indexer,
+          state,
+          data,
+        },
+      },
+    }
+  );
```
Comment (wliyongfeng) [inline packages/scan/src/events/treasury/tip.js:84]: Do tip update in another function.

**Comment by hyifeng** [comment]
ref #24

**Pull request merged**